src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java

```diff
@@ -0,0 +1,171 @@
+import de.tum.in.www1.artemis.domain.LearningObject;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.competency.LearningPath;
+import de.tum.in.www1.artemis.domain.lecture.LectureUnit;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.LearningPathRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastInstructor;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastStudent;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.LearningPathService;
+import de.tum.in.www1.artemis.web.rest.dto.PageableSearchDTO;
+import de.tum.in.www1.artemis.web.rest.dto.SearchResultPageDTO;
+import de.tum.in.www1.artemis.web.rest.dto.learningpath.LearningPathRecommendation;
+import de.tum.in.www1.artemis.web.rest.dto.learningpath.NgxLearningPathDTO;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+
+@RestController
+@RequestMapping("/api")
+public class LearningPathResource {
+
+    private final Logger log = LoggerFactory.getLogger(LearningPathResource.class);
+
+    private final CourseRepository courseRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final UserRepository userRepository;
+
+    private final LearningPathService learningPathService;
+
+    private final LearningPathRepository learningPathRepository;
+
+    public LearningPathResource(CourseRepository courseRepository, AuthorizationCheckService authorizationCheckService, UserRepository userRepository,
+            LearningPathService learningPathService, LearningPathRepository learningPathRepository) {
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.userRepository = userRepository;
+        this.learningPathService = learningPathService;
+        this.learningPathRepository = learningPathRepository;
+    }
+
+    /**
+     * PUT /courses/:courseId/learning-paths/enable : Enables and generates learning paths for the course
+     *
+     * @param courseId the id of the course for which the learning paths should be enabled
+     * @return the ResponseEntity with status 200 (OK) and with body the updated course
+     */
+    @PutMapping("/courses/{courseId}/learning-paths/enable")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Course> enableLearningPathsForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to enable learning paths for course with id: {}", courseId);
+        Course course = courseRepository.findWithEagerCompetenciesByIdElseThrow(courseId);
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
```

Comment:
```suggestion
        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
```

Also applies to all other occurrences where you don't need the user